File: cs50/sentimental-readability/readability.py
from cs50 import get_string


def main():

    text = get_string("Text: ")

    L = count_letters(text) / count_words(text) * 100
    S = count_sentences(text) / count_words(text) * 100

    grade = round(0.0588 * L - 0.29 * S - 15.8)

    if (grade >= 16):
        print("Grade 16+")

    elif (grade < 1):
        print("Before Grade 1")

    else:
        print("Grade", grade)

def count_letters(text):
    letters = 0

    for i in range(len(text)):
        if (text[i].isalpha()):
            letters += 1
    return letters

def count_words(text):
    words = 1

    for i in range(len(text)):
        if (text[i].isspace()):
            words += 1

    return words

def count_sentences(text):
    sentences = 0

    for i in range(len(text)):
        if (text[i] == '.' or text[i] == '?' or text[i] == '!'):
            sentences += 1

    return sentences

if __name__ == "__main__":
    main()

File: cs50/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");

    // Calculates Grade with formula. Uses the functions to get letters, words, and sentences
    int grade = round(0.0588 * (((float)count_letters(text) / (float)count_words(text)) * 100.0) - 0.296 * (((float)count_sentences(
                          text) / (float)count_words(text)) * 100.0) - 15.8);

    // Checks the grade
    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }

    return 0;
}

int count_letters(string text)
{
    int letters = 0;

    // Counts the number of letters
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters++;
        }
    }

    return letters;
}

int count_words(string text)
{
    // Assumpation: will contain at least one word
    // will not start or end with a space
    // will not have multiple spaces in a row
    int words = 1;

    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }

    return words;
}

int count_sentences(string text)
{
    int sentences = 0;

    // Counts the number of sentences
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }

    return sentences;
}